### PR TITLE
Update nodejs.md

### PR DESCRIPTION
**Pull request by elee0625**
### What does this PR do?
Node Public Doc PR

### Motivation
Part of Onboarding Improvement Initiative 

### Additional Notes
Bit more changes here compared to other languages.

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/tracing/trace_collection/dd_libraries/nodejs.md [hunk 1]
```diff
@@ -29,11 +29,82 @@ further_reading:
 ---
 ## Compatibility requirements
 
-The latest version of the NodeJS Tracer officially supports versions `>=12`. Versions `8` and `10` are supported in maintenance mode on the `0.x` release line. For more information about our Node version support and the supported versions, see the [Compatibility Requirements][1] page.
+The latest NodeJS Tracer supports versions `>=12`. For a full list of Datadog’s Node version and framework support (including legacy and maintenance versions), see the [Compatibility Requirements][1] page.
```
Comment (kayayarai) [inline content/en/tracing/trace_collection/dd_libraries/nodejs.md:32]:
```suggestion
The latest Node.js Tracer supports versions `>=12`. For a full list of Datadog’s Node.js version and framework support (including legacy and maintenance versions), see the [Compatibility Requirements][1] page.
```

File: content/en/tracing/trace_collection/dd_libraries/nodejs.md [hunk 2]
```diff
@@ -29,11 +29,82 @@ further_reading:
...
+{{% tab "Containers" %}}
+
+1. Set `apm_non_local_traffic: true` in the `apm_config` section of your main [`datadog.yaml` configuration file][1].
+
+2. See the specific setup instructions to ensure that the Agent is configured to receive traces in a containerized environment:
+
+{{< partial name="apm/apm-containers.html" >}}
+</br>
+
+3. After having instrumented your application, the tracing client sends traces to `localhost:8126` by default.  If this is not the correct host and port change it by setting the below env variables:
```
Comment (kayayarai) [inline content/en/tracing/trace_collection/dd_libraries/nodejs.md:58]:
```suggestion
3. The tracing client sends traces to `localhost:8126` by default. If this is not the correct host and port for your Agent, set the `DD_AGENT_HOST` and `DD_TRACE_AGENT_PORT` environment variables by running:
```

File: content/en/tracing/trace_collection/dd_libraries/nodejs.md [hunk 3]
```diff
@@ -29,11 +29,82 @@ further_reading:
...
+
+    `DD_AGENT_HOST` and `DD_TRACE_AGENT_PORT`.
```
Comment (kayayarai) [inline content/en/tracing/trace_collection/dd_libraries/nodejs.md:60]:
```suggestion
```

File: content/en/tracing/trace_collection/dd_libraries/nodejs.md [hunk 4]
```diff
@@ -29,11 +29,82 @@ further_reading:
...
+
+    ```sh
+    DD_AGENT_HOST=<HOSTNAME> DD_TRACE_AGENT_PORT=<PORT> node server
+    ```
+
+    To use a different protocol such as UDS, specify the entire URL as a single ENV variable `DD_TRACE_AGENT_URL`.
```
Comment (kayayarai) [inline content/en/tracing/trace_collection/dd_libraries/nodejs.md:66]:
```suggestion
   To use a different protocol such as UDS, specify the entire URL as a single environment variable, `DD_TRACE_AGENT_URL`.
```

File: content/en/tracing/trace_collection/dd_libraries/nodejs.md [hunk 5]
```diff
@@ -29,11 +29,82 @@ further_reading:
...
+To set up Datadog APM in AWS Lambda, see the [Tracing Serverless Functions][1] documentation.
+
+
+[1]: /tracing/serverless_functions/
+{{% /tab %}}
+{{% tab "Other Environments" %}}
+
+Tracing is available for a number of other environments, such as  [Heroku][1], [Cloud Foundry][2], [AWS Elastic Beanstalk][3], and [Azure App Service][4].
+
+For other environments, please refer to the [Integrations][5] documentation for that environment and [contact support][6] if you are encountering any setup issues.
```
Comment (kayayarai) [inline content/en/tracing/trace_collection/dd_libraries/nodejs.md:91]:
```suggestion
For other environments, see the [Integrations][5] documentation for that environment and [contact support][6] if you encounter setup issues.
```

File: content/en/tracing/trace_collection/dd_libraries/nodejs.md [hunk 6]
```diff
@@ -29,11 +29,82 @@ further_reading:
...
+{{% tab "AWS Lambda" %}}
+
+To set up Datadog APM in AWS Lambda, see the [Tracing Serverless Functions][1] documentation.
+
+
+[1]: /tracing/serverless_functions/
+{{% /tab %}}
+{{% tab "Other Environments" %}}
+
+Tracing is available for a number of other environments, such as  [Heroku][1], [Cloud Foundry][2], [AWS Elastic Beanstalk][3], and [Azure App Service][4].
```
Comment (kayayarai) [inline content/en/tracing/trace_collection/dd_libraries/nodejs.md:89]:
```suggestion
Tracing is available for other environments, including [Heroku][1], [Cloud Foundry][2], [AWS Elastic Beanstalk][3], and [Azure App Service][4].
```

File: content/en/tracing/trace_collection/dd_libraries/nodejs.md [hunk 7]
```diff
@@ -29,11 +29,82 @@ further_reading:
...
+[2]: /integrations/cloud_foundry/#trace-collection
+[3]: /integrations/amazon_elasticbeanstalk/
+[4]: /infrastructure/serverless/azure_app_services/#overview
+[5]: /integrations/
+[6]: /help/
+{{% /tab %}}
+{{< /tabs >}}
+
+
+See the [tracer settings][3] for the list of initialization options.
```
Comment (kayayarai) [inline content/en/tracing/trace_collection/dd_libraries/nodejs.md:103]:
```suggestion
Read [tracer settings][3] for a list of initialization options.
```

File: content/en/tracing/trace_collection/dd_libraries/nodejs.md [hunk 8]
```diff
@@ -29,11 +29,82 @@ further_reading:
...
+
+### Instrument Your Application
+
+After the agent is installed, follow these steps to add the Datadog tracing library to your Node.js applications:
```
Comment (kayayarai) [inline content/en/tracing/trace_collection/dd_libraries/nodejs.md:107]:
```suggestion
After the Agent is installed, follow these steps to add the Datadog tracing library to your Node.js applications:
```

File: content/en/tracing/trace_collection/dd_libraries/nodejs.md [hunk 9]
```diff
@@ -29,11 +29,82 @@ further_reading:
...
+[4]: /infrastructure/serverless/azure_app_services/#overview
+[5]: /integrations/
+[6]: /help/
+{{% /tab %}}
+{{< /tabs >}}
+
+
+See the [tracer settings][3] for the list of initialization options.
+
+### Instrument Your Application
```
Comment (kayayarai) [inline content/en/tracing/trace_collection/dd_libraries/nodejs.md:105]:
```suggestion
### Instrument your application
```

File: content/en/tracing/trace_collection/dd_libraries/nodejs.md (hunk 1 above)
Comment (bengl) [inline content/en/tracing/trace_collection/dd_libraries/nodejs.md:32]:
```suggestion
The latest Node.js Tracer supports versions `>=12`. For a full list of Datadog’s Node.js version and framework support (including legacy and maintenance versions), see the [Compatibility Requirements][1] page.
```

Always Node.js. Never NodeJS or Node.

File: content/en/tracing/trace_collection/dd_libraries/nodejs.md [hunk 10]
```diff
@@ -56,10 +127,6 @@ To add the Datadog tracing library to your Node.js applications, follow these st
 
    When using a transpiler such as TypeScript, Webpack, Babel, or others, import and initialize the tracer library in an external file and then import that file as a whole when building your application.
 
-3. [Configure the Datadog Agent for APM](#configure-the-datadog-agent-for-apm)
-
-4. Add any desired [configuration](#configuration) to the tracer, such as `service`, `env`, and `version` for [Unified Service Tagging][2].
-
 ### Adding the tracer in code
 
 ##### JavaScript
```
Comment (kayayarai) [inline content/en/tracing/trace_collection/dd_libraries/nodejs.md:132]:
```suggestion
#### JavaScript
```

**Review by bengl — CHANGES_REQUESTED** [review]

**Review by kayayarai — CHANGES_REQUESTED** [review]

File: content/en/tracing/trace_collection/dd_libraries/nodejs.md [hunk 11]
```diff
@@ -29,11 +29,82 @@ further_reading:
...
+
+Follow the [Quickstart instructions][3] within the Datadog app for the best experience, including:
+
+- Step-by-step instructions scoped to your deployment configuration (hosts, Docker, Kubernetes, or Amazon ECS).
+- Dynamically set `service`, `env`, and `version` tags.
+- Enable the Continuous Profiler, ingesting 100% of traces, and Trace ID injection into logs during setup.
+
+### Configure the Datadog Agent for APM
+
+Install and configure the Datadog Agent to receive traces from your now instrumented application. By default the Datadog Agent is enabled in your `datadog.yaml` file under `apm_config` with `enabled: true` and listens for trace traffic at `localhost:8126`. For containerized environments, follow the links below to enable trace collection within the Datadog Agent.
```
Comment (kayayarai) [inline content/en/tracing/trace_collection/dd_libraries/nodejs.md:46]:
```suggestion
Install and configure the Datadog Agent to receive traces from your instrumented application. By default the Datadog Agent is enabled in your `datadog.yaml` file under `apm_config` with `enabled: true` and listens for trace traffic at `localhost:8126`. For containerized environments, follow the links below to enable trace collection within the Datadog Agent.
```

**Review by kayayarai — APPROVED** [review]

**Pull request merged**